build-tools/scripts/TestApks.targets

```diff
@@ -273,6 +273,7 @@
         AddResults="true"
         LabelSuffix="-$(Configuration)$(TestsFlavor)"
         Activity="%(_AllArchives.Activity)" />
+<Exec Command="adb shell run-as %(_AllArchives.Package) cat files/.__override__/grefs.txt" />
```

Comment:
This should probably specify `IgnoreExitCode="True'`, because not all of our tests have been updated to emit GREF logging.

In particular, what started the GREF investigation was *not* `Mono.Android_Tests`, but instead `tests/Xamarin.Forms-Performance-Integration`.  Attempting to grab the GREF log for `Mono.Android_Tests` will thus not be at all useful: we're not generating them in the first place (no changes to `src/Mono.Android/Test` to enable GREF logging), and that's not the app that we saw the failure on either.

Reply:
This command should probably *redirect* the output of `cat files/...` to a new file, otherwise the output is just going to go into the build log, which will not be as helpful for later investigation.

The new file will need to be added to the archive so that it can be viewed.

```xml
    <Exec
        IgnoreExitCode="True"
        Command="adb shell run-as %(_AllArchives.Package) cat files/.__override__/grefs.txt > &quot;%(_AllArchives.Package)-%(_AllArchives.Package)-$(Configuration)$(TestsFlavor)-grefs.txt&quot;"
    />
```

Note that the directory that the `*-grefs.txt` file is written to is not specified, and offhand I'm not sure where it should be placed to make it easier to include in the uploaded archives.